src/SetupApi.Desktop/SetupApi+SP_DEVICE_INTERFACE_DETAIL_DATA.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Runtime.InteropServices;
+
+    /// <content>
+    /// Contains the <see cref="SP_DEVICE_INTERFACE_DETAIL_DATA"/> nested type.
+    /// </content>
+    public partial class SetupApi
+    {
+        /// <summary>
+        /// An SP_DEVICE_INTERFACE_DETAIL_DATA structure contains the path for a device interface.
+        /// </summary>
+        [StructLayout(LayoutKind.Sequential)]
```

Comment:
I didn't test with strings actually, just suspected that it would act the same.

I was thinking about fixed-length strings but reflecting on it even there it wouldn't work as the CLR representation is different than a `char*` and would force a marshaling.

So no in fine I think that it would only work for embedded char (directly or via fixed arrays) only.

It work there because the default is Ansi (So marshalling is required) and switching to Unicode make the the marshaller know that the structure is blittable.
Once it know that it consider that a pointer can safely be used directly from the managed world.